nautobot/core/celery/__init__.py

```diff
@@ -66,19 +67,47 @@ def register_task(self, task, **options):
 
 
 @signals.import_modules.connect
-def import_tasks_from_jobs_root(sender, **kwargs):
-    """Load jobs from JOBS_ROOT on celery workers."""
+def import_jobs_as_celery_tasks(sender, **kwargs):
+    """
+    Import system Jobs into Celery as well as Jobs from JOBS_ROOT and GIT_ROOT.
+
+    Note that app-provided Jobs are automatically imported at startup time via NautobotAppConfig.ready()
+    """
+    logger.debug("Importing system Jobs")
+    sender.loader.import_task_module("nautobot.core.jobs")
+
     jobs_root = settings.JOBS_ROOT
     if jobs_root and os.path.exists(jobs_root):
         if jobs_root not in sys.path:
             sys.path.append(jobs_root)
         for _, module_name, _ in pkgutil.iter_modules([jobs_root]):
             try:
+                logger.debug("Importing Jobs from %s in JOBS_ROOT", module_name)
                 sender.loader.import_task_module(module_name)
             except Exception as exc:
                 # logger.error(f"Unable to load module '{module_name}' from {jobs_root}: {exc:}")
                 logger.exception(exc)
 
+    git_root = settings.GIT_ROOT
+    if git_root and os.path.exists(git_root):
+        if git_root not in sys.path:
+            sys.path.append(git_root)
+
+        from nautobot.extras.datasources.git import ensure_git_repository
+
+        django_apps = kwargs.get("apps", global_apps)
```

Comment:
It doesn't look like you're passing in `apps` anywhere this is being used. If this is staying as an optional kwarg please add it to the docstring.

Reply:
Yeah, this was an early-morning attempt at fixing the issue with accessing the DB during a `ready()` function. I've reworked this locally to be more correct, new diffs will remove this.